ios/IoReactNativeJwt.swift

```diff
@@ -151,6 +181,76 @@ class IoReactNativeJwt: NSObject {
             reject("Error", "Unable to parse string to hash", nil);
         }
     }
+    
+    @objc
+    func enc(_ plaintext: String, header: NSDictionary, jwk: NSDictionary, resolver resolve: RCTPromiseResolveBlock, rejecter reject: RCTPromiseRejectBlock) {
+        do {
+            let publicKeyJson = try JSONSerialization.data(withJSONObject: jwk, options:[] )
+            let message = plaintext.data(using: .utf8)!
+        
+            if let content = header as? [String:AnyObject] {
+                let jweHeader = try JWEHeader(parameters: content)
+                let payload = Payload(message)
+                
+                if isECKey(jwk:jwk) {
+                    reject("Error", "EC not supported", nil);
+                } else {
+                    let rsaJwk = try RSAPublicKey(data: publicKeyJson)
+                    let publicKey = try rsaJwk.converted(to: SecKey.self)
+                    let encrypter = Encrypter(keyManagementAlgorithm: try getKeyManagmentAlg(header: header), contentEncryptionAlgorithm: try getContentEncryptionAlgorithm(header: header), encryptionKey: publicKey)!
+                    
+                    let jwe = try JWE(header: jweHeader, payload: payload, encrypter: encrypter)
+
+                    resolve(jwe.compactSerializedString)
+                    
+                }
+ 
+            } else {
+                reject("Error", "Header is invalid", nil);
+            }
+        }
+        
+        catch {
+            reject("Error", "\(error)", error);
+        }
+
+
+    }
+    /*
+    @objc
+    func dec(_ token: String, header: NSDictionary, jwk: NSDictionary, resolver resolve: RCTPromiseResolveBlock, rejecter reject: RCTPromiseRejectBlock) {
+        do {
+            let privateKeyJson = try JSONSerialization.data(withJSONObject: jwk, options:[] )
+           
+            let jwe = try JWE(compactSerialization: token)
+            if isECKey(jwk:jwk) {
+                reject("Error", "EC not supported", nil);
+            } else {
+                let rsaJwk = try RSAPrivateKey(data: privateKeyJson)
+                let privateKey = try rsaJwk.converted(to:  SecKey.self)
+                let decrypter = Decrypter(keyManagementAlgorithm: try getKeyManagmentAlg(header: header), contentEncryptionAlgorithm: try getContentEncryptionAlgorithm(header: header), decryptionKey: rsaJwk)
+                if(decrypter != nil){
+                    let decrypted = try jwe.decrypt(using: decrypter!)
+                    let message = String(data: decrypted.data(), encoding: .utf8)!
+                    resolve(message)
+                } else {
+                    reject("Error", "Invalid decrypter", nil);
+                }
+                
+            }
+
+         
+        }
+        
+        catch let error as NSError {
+            reject("Error", "\(error)", error);
+        }
+
+
+    }
+     */
```

Comment:
Why is this commented out?

Reply:
It might make sense instead of commenting out this method to return an error like: "iOS does not support ....."?

Reply:
Unfortunately android is not supported either. 😞